pkg/serdes/double_deserialization_provider.go

```diff
@@ -13,8 +13,8 @@ func (DoubleDeserializationProvider) LoadSchema(_ string, _ map[string]string) e
 }
 
 func (DoubleDeserializationProvider) Deserialize(data []byte) (string, error) {
-	if len(data) == 0 {
-		return "", nil
+	if len(data) != 8 {
+		return "", fmt.Errorf("the double key is invalid")
```

Comment:
This is a breaking change. Let's keep the original `if` statement and put it above this one.